pkg/graph/resolvers/key_contact_category.go

```diff
@@ -0,0 +1,137 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+
+	"github.com/cms-enterprise/mint-app/pkg/authentication"
+	"github.com/cms-enterprise/mint-app/pkg/models"
+	"github.com/cms-enterprise/mint-app/pkg/sqlutils"
+	"github.com/cms-enterprise/mint-app/pkg/storage"
+)
+
+// CreateKeyContactCategory creates a new key contact category.
+// It inserts a new category record with the provided category name.
+func CreateKeyContactCategory(ctx context.Context, logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	categoryStr string,
+) (*models.KeyContactCategory, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	newCategory := models.NewKeyContactCategory(principalAccount.ID, categoryStr)
+	createdCategory, err := storage.KeyContactCategoryCreate(store, logger, newCategory)
+	if err != nil {
+		return nil, err
+	}
+
+	return createdCategory, nil
+}
+
+// UpdateKeyContactCategory updates an existing key contact category.
+// Only category field can be changed. Returns the updated category.
+func UpdateKeyContactCategory(ctx context.Context, logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	id uuid.UUID,
+	changes map[string]interface{},
+) (*models.KeyContactCategory, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	existingCategory, err := storage.KeyContactCategoryGetByID(store, logger, id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get category with id %s: %w", id, err)
+	}
+
+	if existingCategory == nil {
```

Comment:
nitpick: technically checking if existingCategory is nil here is redundant since that should be caught in the err above, before removing this check though it would be good to verify that the err works as intended